test/test_pymbolic.py

```diff
@@ -719,6 +818,48 @@ def map_spatial_constant(self, expr):
     assert MyMapper()(c) == 2*c
     assert IdentityMapper()(c) == c
 
+# }}}
+
+
+# {{{ test_equality_complexity
+
+def test_equality_complexity():
+    # NOTE: https://github.com/inducer/pymbolic/issues/73
+    from numpy.random import default_rng
+
+    def construct_intestine_graph(depth=64, seed=0):
+        rng = default_rng(seed)
+        x = prim.Variable("x")
+
+        for _ in range(depth):
+            coeff1, coeff2 = rng.integers(1, 10, 2)
+            x = coeff1 * x + coeff2 * x
+
+        return x
+
+    def check_equality():
+        graph1 = construct_intestine_graph()
+        graph2 = construct_intestine_graph()
+        graph3 = construct_intestine_graph(seed=3)
+
+        assert graph1 == graph2
+        assert graph2 == graph1
+        assert graph1 != graph3
+        assert graph2 != graph3
+
+    # NOTE: this should finish in a second!
+    import multiprocessing
+    p = multiprocessing.Process(target=check_equality)
+    p.start()
+    p.join(timeout=1)
+    with pytest.raises(AssertionError):
```

Comment:
It might be better to do a "proper" xfail: https://docs.pytest.org/en/6.2.x/skipping.html#xfail-mark-test-functions-as-expected-to-fail

Reply:
Sure thing! Added in 
https://github.com/inducer/pymbolic/compare/a9a56d46abf59f6f93ded6204a91f578dbbd8c07..d2364df109f03c6f898abd35a132c5b64541960e